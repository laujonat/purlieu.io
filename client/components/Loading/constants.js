export const WELCOME_DESCRIPTION =
  "Welcome to Purleiu.io, an orthogonal projection of boundaries for ride sharing applications. Enter a dollar amount between $10 - $400 into the form above and we'll calculate a matrix displaying how far you can go in your next Lyft ride request!"
