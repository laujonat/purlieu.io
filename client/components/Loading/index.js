import React from "react"
import styled from "styled-components"

export const LoadingText = styled.div`
  display: none;
  justify-content: center;

  ${({ active }) =>
    active &&
    `
    color: blue;
    display: block;
  `}
`
