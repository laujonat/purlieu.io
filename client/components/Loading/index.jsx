import React from "react"

export const Loading = ({ children, customClasses }) => (
  <div id="loading-overlay">
    <div className={`loading-content ${customClasses}`}>{children}</div>
  </div>
)
