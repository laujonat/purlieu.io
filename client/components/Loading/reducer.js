import {
  RECEIVE_CLIENT_LOCATION,
  RECEIVE_CLIENT_LOCATION_SUCCESS,
  RECEIVE_CLIENT_LOCATION_ERROR,
  RECEIVE_MARKER_LOCATION,
  RECEIVE_MARKER_LOCATION_SUCCESS,
  RECEIVE_MARKER_LOCATION_ERROR,
  DRAW_POLYGON,
  DRAW_POLYGON_SUCCESS,
  DRAW_POLYGON_ERROR
} from "../Map/actions"

import {
  RECEIVE_BOUNDARIES,
  RECEIVE_BOUNDARIES_SUCCESS,
  RECEIVE_BOUNDARIES_ERROR
} from "../../boundaries/actions"

const initialState = {}

export const reducer = (state = initialState, action) => {
  Object.freeze(state)
  const newState = { ...state }

  switch (action.type) {
    case DRAW_POLYGON:
    case RECEIVE_BOUNDARIES:
    case RECEIVE_MARKER_LOCATION:
    case RECEIVE_CLIENT_LOCATION:
      return {
        ...newState,
        [action.type]: true
      }
    case DRAW_POLYGON_SUCCESS:
    case DRAW_POLYGON_ERROR:
      delete newState[DRAW_POLYGON]
      return newState
    case RECEIVE_BOUNDARIES_SUCCESS:
    case RECEIVE_BOUNDARIES_ERROR:
      delete newState[RECEIVE_BOUNDARIES]
      return newState
    case RECEIVE_MARKER_LOCATION_SUCCESS:
    case RECEIVE_MARKER_LOCATION_ERROR:
      delete newState[RECEIVE_MARKER_LOCATION]
      return newState
    case RECEIVE_CLIENT_LOCATION_SUCCESS:
    case RECEIVE_CLIENT_LOCATION_ERROR:
      delete newState[RECEIVE_CLIENT_LOCATION]
      return newState
    default:
      return state
  }
}

export default reducer
