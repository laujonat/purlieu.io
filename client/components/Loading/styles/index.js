import styled from "styled-components"
import { spaces, fonts } from "../../../lib/styles"

export const Container = styled.div`
  visibility: hidden;
  justify-content: center;
  font-size: ${fonts.small};
  height: ${spaces.xxSmall};
  margin-top: ${spaces.xxSmall};

  ${({ active }) =>
    active &&
    `
    visibility: visible;
    color: blue;
    display: flex;
  `}
`
