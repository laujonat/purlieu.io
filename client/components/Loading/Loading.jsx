import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { spaces, fonts } from "../../lib/styles"

const Container = styled.div`
  visibility: hidden;
  justify-content: center;
  font-size: ${fonts.small};
  height: ${spaces.xxSmall};
  margin-top: ${spaces.xxSmall};

  ${({ active }) =>
    active &&
    `
    visibility: visible;
    color: blue;
    display: flex;
  `}
`

export const Loading = ({ active }) => <Container active={active}>Loading..</Container>

Loading.defaultProps = {
  active: false
}

Loading.propTypes = {
  active: PropTypes.bool
}
