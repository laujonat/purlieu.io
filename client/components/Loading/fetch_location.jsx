import React, { Fragment } from "react"
import Loading from "."

export default class FetchLocationForm extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const { currentStatus } = this.props

    return (
      <Fragment>
        <Loading customClasses="fetch-location">
          <img
            className="fetch-location-img"
            src="https://i.imgur.com/P0CmA6f.png"
          />
          <p className="status">{currentStatus}</p>
          <p className="wait-text">Please wait...</p>
        </Loading>
      </Fragment>
    )
  }
}
