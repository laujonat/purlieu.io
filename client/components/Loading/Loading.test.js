import React from "react"
import { Loading } from "./Loading"
import { shallow } from "enzyme"

describe("Loading", () => {
  let props
  let wrapper

  it("renders without crashing", () => {
    props = {
      isFetching: true
    }
    wrapper = shallow(<Loading active={props.isFetching} />)
    expect(wrapper).toBeDefined()
  })

  it("renders and displayed when active is true", () => {
    props = {
      isFetching: true
    }
    wrapper = shallow(<Loading active={props.isFetching} />)
    expect(wrapper.find(Loading)).toHaveLength(1)
  })

  it("renders and not displayed when active is false", () => {
    props = {
      isFetching: false
    }
    wrapper = shallow(<Loading active={props.isFetching} />)
    expect(wrapper.find(Loading)).toHaveLength(0)
  })
})
