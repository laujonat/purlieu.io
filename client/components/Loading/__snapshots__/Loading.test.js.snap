// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders and displayed when active is true 1`] = `
<div
  className="sc-bdVaJa klLhBI"
>
  Loading..
</div>
`;

exports[`Loading renders and not displayed when active is false 1`] = `
<div
  className="sc-bdVaJa bJFeRj"
>
  Loading..
</div>
`;

exports[`Loading renders without crashing 1`] = `
<div
  className="sc-bdVaJa bJFeRj"
>
  Loading..
</div>
`;
